src/LibraryManager.Vsix/Contracts/Logger.cs

```diff
@@ -91,110 +91,154 @@ private static bool EnsurePane()
-                    LogEvent(Environment.NewLine + summaryCancellationText + Environment.NewLine, LogLevel.Operation);
+        private static string GetAllFailuresString(OperationType operation, int totalCount, string libraryId)
+        {
+            switch (operation)
+            {
+                case OperationType.Restore:
+                    {
+                        return string.Format(LibraryManager.Resources.Text.Restore_NumberOfLibrariesFailed, totalCount);
+                    }
+                case OperationType.Clean:
+                    {
+                        return string.Format(LibraryManager.Resources.Text.Clean_NumberOfLibrariesFailed, totalCount);
+                    }
+                case OperationType.Uninstall:
+                    {
+                        return string.Format(LibraryManager.Resources.Text.Uninstall_LibraryFailed, libraryId ?? string.Empty);
+                    }
+                case OperationType.Upgrade:
+                    {
+                        return string.Format(LibraryManager.Resources.Text.Update_LibraryFailed, libraryId ?? string.Empty);
+                    }
+                default:
+                    {
+                        return string.Empty;
+                    }
+            }
+        }
 
-                    foreach (var result in cancelledRestores)
+        private static string GetAllCancelledString(OperationType operation, int totalCount, TimeSpan timeSpan, string libraryId)
+        {
+            switch (operation)
+            {
+                case OperationType.Restore:
                     {
-                        var cancellationText = operation == OperationType.Clean ?
-                            string.Format(LibraryManager.Resources.Text.Clean_OperationCancelled, result.InstallationState.LibraryId):
-                            string.Format(LibraryManager.Resources.Text.Restore_OperationCancelled, result.InstallationState.LibraryId);
-                        LogEvent(cancellationText, LogLevel.Operation);
+                        return string.Format(LibraryManager.Resources.Text.Restore_NumberOfLibrariesCancelled, totalCount, Math.Round(timeSpan.TotalSeconds, 2));
+                    }
+                case OperationType.Clean:
+                    {
+                        return string.Format(LibraryManager.Resources.Text.Clean_NumberOfLibrariesCancelled, totalCount, Math.Round(timeSpan.TotalSeconds, 2));
+                    }
+                case OperationType.Uninstall:
+                    {
+                        return string.Format(LibraryManager.Resources.Text.Uninstall_LibraryCancelled, libraryId ?? string.Empty);
+                    }
+                case OperationType.Upgrade:
+                    {
+                        return string.Format(LibraryManager.Resources.Text.Update_LibraryCancelled, libraryId ?? string.Empty);
+                    }
+                default:
+                    {
```

Comment:
This switch's formatting doesn't match that in the following method.

Reply:
Fixed